test/test_lib.py

```diff
@@ -547,15 +557,15 @@ def _test_submodule(self, *, relative: bool = False):
         assert baz.LIBAPI == 2
         assert baz.quux.this == 42
 
-    def test_submodule_absolute(self):
-        self._test_submodule(relative=False)
+    def test_submodule_absolute(self, tmp_path: pathlib.Path, monkeypatch: pytest.MonkeyPatch):
+        self._test_submodule(tmp_path, monkeypatch, relative=False)
 
-    def test_submodule_relative(self):
-        self._test_submodule(relative=True)
+    def test_submodule_relative(self, tmp_path: pathlib.Path, monkeypatch: pytest.MonkeyPatch):
+        self._test_submodule(tmp_path, monkeypatch, relative=True)
 
-    def test_others_found(self):
-        tmpdir = self._mkdtemp()
-        sys.path = [tmpdir]
+    def test_others_found(self, tmp_path: pathlib.Path, monkeypatch: pytest.MonkeyPatch):
+        tmpdir = str(tmp_path)
+        monkeypatch.syspath_prepend(tmpdir)  # type: ignore
```

Comment:
It seems like this is a functional change - before, we would *only* look in `tmpdir`, and now we will look in `tmpdir` and also whatever else is already in sys.path. I'm not sure if this is an issue or not - do you know?